### PR TITLE
Es score

### PR DESCRIPTION
**Pull request by lyrated**
- some restructuring on userController, jobController and matchesController (e.g. moved indexJobOffers method from userC to jobC, separated some methods in matchesC to reuse them inside userC and searchC)
- job cards changed a bit and added a button to add jobs on jobs/index.ejs
- for candidate and job offer: new field "max_score"
- creating a new account or adding a job will calculate the maximum score and add it to max_score
- seed.js also seeds jobs with soft skills now and max_score
- a score is calculated based on its percentage of max_score 
- compatibility "Excellent", "Great", "Good" is determined based on the score percentage
- differently filled circles show the compatibility in matches 
- bad matches are filtered out
- "N/A" and grey circles displayed when a score is missing

**IMPORTANT**: Start Elasticsearch and create a candidate or a job offer to generate a max_score to see the compatibility.

### CODE REVIEW

**Comment by kimc0de** [comment]
Hey thank youuu great job <3 the front end look great. 

I tested with the updated seed and created candidate and recruiter the first result of matches were good, but when I edit the profile of the candidate: desire position from Frontend Developer to Backend Developer and edit the tech stack to backend tech stack, the first match was still the frontend developer... and then the others below were full-stack, nothing about backend developer is shown. In our query, if I'm not mistaken the matches must match job title, right? So maybe this's a bug..?

Screenshot 2021-01-24 at 16 23 38

Screenshot 2021-01-24 at 16 23 45

**Comment by lyrated** [comment]
> I tested with the updated seed and created candidate and recruiter the first result of matches were good, but when I edit the profile of the candidate: desire position from Frontend Developer to Backend Developer and edit the tech stack to backend tech stack, the first match was still the frontend developer... and then the others below were full-stack, nothing about backend developer is shown. In our query, if I'm not mistaken the matches must match job title, right? So maybe this's a bug..?

thanks for testing! Yes this doesn't seem right... can you try on Kibana the same query and see if it also gets those results? Maybe update the work_culture_keywords too... Even if that is an error, I don't know how to fix it since we are not doing the matching algorithm but Elasticsearch does it :(

**Comment by ellakatajisto** [comment]
Heyy looks amazing! The visualization of the score looks so professional 🤩 i did what you suggested on slack.
Remarks (some aren't relevant to your work):
- when i try to click "more" on the matches of users i've created myself, i get this error. it doesn't happen when i click on "more" on the seeded matches. it happens both with job offers and candidates (i restarted elasticsearch a couple times so i hope it wasn't my mistake):
Näyttökuva 2021-1-24 kello 20 23 26

Näyttökuva 2021-1-24 kello 20 14 58
- also for some reason "company" of the job offer isn't being shown on the match card for candidate
- I almost write a long text about the underscore problem with the work culture keywords (not related to your work) but then saw that Kim had already fixed that error :D

File: controllers/matchesController.js [hunk 1]
```diff
@@ -69,130 +57,168 @@ module.exports = {
 
     Job.findById(jobId).then(job => {
       // sort the keywords by importance
-      let sortedHardSkills = getSortedKeywords("hard_skills.name", job.hard_skills);
-      let sortedSoftSkills = getSortedKeywords("soft_skills.name", job.soft_skills);
+      let sortedHardSkills = module.exports.getSortedKeywords("hard_skills.name", job.hard_skills);
```
Comment (mbareeva) [inline controllers/matchesController.js:60]: just wondeting, why we need module.exports here if this function is in the block module.exports = {...}? IT does not work like this, if we require matchesController  in another class and call on this?

**Review by mbareeva — COMMENTED** [review]
thank you for work! code review done, the ui and functionality will check afterwards. that is exactly how I was imagining this score thingy on FE!!

**Comment by lyrated** [comment]
>     * when i try to click "more" on the matches of users i've created myself, i get this error. it doesn't happen when i click on "more" on the seeded matches. it happens both with job offers and candidates (i restarted elasticsearch a couple times so i hope it wasn't my mistake):

Yeah this is weird... if that value in your screenshot is the ID, then it doesn't look like MongoDB ID (which would usually start with a number) .... dunno, I tried it with my database and couldn't get this error :/

> 
>     * also for some reason "company" of the job offer isn't being shown on the match card for candidate

I have noticed this error for a while now :D I put it in the bug ticket but tbh I could just fix it now that I looked into it (wrong variable name was passed in the ejs)

File: controllers/jobController.js [hunk 2]
```diff
@@ -58,23 +81,23 @@ module.exports = {
         let jobId = req.params.jobId;
         let user = req.user;
 
-        Job.findOneAndRemove({_id: jobId})
-            .then((job) =>{    
+        Job.findOneAndRemove({ _id: jobId })
+            .then((job) => {
                 console.log('user to update:', user);
```
[the remark this answers is not in the document]
Comment (lyrated) [reply controllers/jobController.js:86]: just curious: you want ALL logs removed (except error logs)? Or should we keep some for information/confirmation?

File: controllers/matchesController.js (hunk 1 above)
Comment (lyrated) [reply controllers/matchesController.js:60]: Yeah whenever I tried to call a function from module.exports of the same file, it always said it's not a function. That's why in the beginning of matchesController, we had all those helper functions _outside_ of module exports. But after some googling I found out that you had to call it like that -> `module.exports.XXX()` when it's from the same controller :) and outside of it you just import the controller in a variable and call it like `matchesController.XXX()`

**Review by joshuameiser — APPROVED** [review]
I might need an introduction of all the changes to fully understand what has been changed in the code (or rather for what reason, such as async/await). That being said, from what I understand it looks good. 

I also pulled the branch and tested the scoring as instructed. Everything worked without a hitch, as long as I create new users. Great work!

**Pull request merged**